### ISSUE TITLE
mb_strlen() expects parameter 1 to be string, array given (View: /home/xxx/vendor/tcg/voyager/resources/views/bread/browse.blade.php)

### ISSUE THREAD

**Issue by mforcer** [thedevdojo/voyager#3420]
## Version information

 - Laravel: v5.6.28
 - Voyager: v1.1.3
 - PHP: 7.1
 - Database: Maria DB 10.1

## Description

Whenever I try to browse some BREAD items I get the following error:

`mb_strlen() expects parameter 1 to be string, array given (View: /home/eatnaked/vendor/tcg/voyager/resources/views/bread/browse.blade.php)`

This seems to be fairly common but mainly centred around users/settings/locale or something that is mentioned in other bugs but this doesn't appear to be the case here... I've added some tables in BREAD and when trying to view them the above error is what I get.

I should note I can browse users just fine.

```
Line 154:                                                     <div class="readmore"><?php echo e(mb_strlen( $data->{$row->field} ) > 200 ? mb_substr($data->{$row->field}, 0, 200) . ' ...' : $data->{$row->field}); ?></div>
```

<img width="798" alt="screen shot 2018-07-25 at 22 40 31" src="https://user-images.githubusercontent.com/850149/43229261-1ee16546-905c-11e8-8eaa-e5d626359499.png">

<img width="1103" alt="screen shot 2018-07-25 at 22 41 09" src="https://user-images.githubusercontent.com/850149/43229211-de7163da-905b-11e8-8c9a-4b56255eeada.png">

**Comment by emptynick** [comment]
Can you find out which field is causing the error (unchecking browse-checkbox, one after another) and tell us which datatype it is plus the raw value from the database?

**Comment by mforcer** [comment]
@emptynick Yes, i've done a little more digging and it's dying on fields that are cast as an array.

e.g.

```
    protected $casts = [
        'field_name' => 'array',
    ];
```

`field_name` is stored in the database as json data.

**Comment by tommiekn** [comment]
Same issue as @gdbytes , cant work with arrays/json, seems declaring  arrays on casts doesnt works

**Comment by emptynick** [comment]
I dont really know how we should display an array or JSON in the browse-view.
Any suggestions?
`Array` `Array of X elements` ...

**Comment by tommiekn** [comment]
I think that first of all you could display it as a text value (without throwing exceptions) so we could actually use the field without having to edit it the db and hide it from voyager.

Would be nice to have a json editor there or something to auto generate and handle fields one day.

**Comment by emptynick** [comment]
I dont think that we will display an array as a string.
Also there is an editor for JSON, but we wont display that in the browse-view of course.

**Comment by tommiekn** [comment]
I have mentioned to display array as a string as temp fix, if the field is visible the app crashes, so would be at least something to be able to see the fields.

Re JSON, there's currently a view to handle json fields in the edit form?

To handle arrays you could also simply show values one per line like in a textarea
index:value
index:value

etc.

In my case, i just need to give the opportunity to the user to input in textarea where each line is an array value.

**Comment by mforcer** [comment]
Anything that prevents the error will do for now.. why not look through it and display it as a simple list?

This is what I did to quickly get around it... I created my own views and conditionally check for the problematic field then apply relevant logic to display the data within the field.

**Comment by tommiekn** [comment]
Hi,
I went ahead an I have wrote a FormFields type called "array_text_area".
If selected this FormFields, a textarea will be shown, array will be processed thru lines so each line is an array value.
Zip with the edits is attached, could someone commit this here as well?

[textarray.zip](https://github.com/the-control-group/voyager/files/2329066/textarray.zip)


TODO
Next step is adding the parsing of "index:value" so it could work an array with indexes as well

Feedback is appreciated,
Thanks

**Comment by luizrobertomeier** [comment]
Any fixes about this? Thanks.
